.github/workflows/wheels-test.yaml

```diff
@@ -71,19 +71,19 @@ jobs:
           export MATRICES="
             pull-request:
               - { ARCH: 'amd64', PY_VER: '3.9', CUDA_VER: '11.8.0', LINUX_VER: 'ubuntu18.04', gpu: 'v100', driver: 'latest' }
-              - { ARCH: 'amd64', PY_VER: '3.10', CUDA_VER: '12.2.2', LINUX_VER: 'ubuntu20.04', gpu: 'v100', driver: 'latest' }
               - { ARCH: 'arm64', PY_VER: '3.10', CUDA_VER: '12.0.1', LINUX_VER: 'ubuntu20.04', gpu: 'a100', driver: 'latest' }
+              - { ARCH: 'amd64', PY_VER: '3.11', CUDA_VER: '12.2.2', LINUX_VER: 'ubuntu20.04', gpu: 'v100', driver: 'latest' }
             nightly:
               - { ARCH: 'amd64', PY_VER: '3.9', CUDA_VER: '11.8.0', LINUX_VER: 'ubuntu18.04', gpu: 'v100', driver: 'latest' }
               - { ARCH: 'amd64', PY_VER: '3.10', CUDA_VER: '11.8.0', LINUX_VER: 'ubuntu18.04', gpu: 'v100', driver: 'latest' }
               - { ARCH: 'arm64', PY_VER: '3.9', CUDA_VER: '11.8.0', LINUX_VER: 'ubuntu20.04', gpu: 'a100', driver: 'latest' }
-              - { ARCH: 'arm64', PY_VER: '3.10', CUDA_VER: '11.8.0', LINUX_VER: 'ubuntu20.04', gpu: 'a100', driver: 'latest' }
+              - { ARCH: 'arm64', PY_VER: '3.11', CUDA_VER: '11.8.0', LINUX_VER: 'ubuntu20.04', gpu: 'a100', driver: 'latest' }
```

Comment:
Let’s reshuffle these a bit. I want one configuration to be amd64 + latest Python + latest CUDA. Currently it seems only ARM has Python 3.11 jobs. I would also tilt the distribution of versions to be evenly weighted, it seems like the matrix tests 3.9 and 3.10 more than 3.11 (I didn’t do an exact count).

Reply:
Thanks for looking @bdice .

This PR *does* currently have a Python 3.11 AMD job for `wheels-test`... but just on PRs.

```text
- { ARCH: 'amd64', PY_VER: '3.11', CUDA_VER: '12.2.2', LINUX_VER: 'ubuntu20.04', gpu: 'v100', driver: 'latest' }
```

Definitely agree the mix across Python versions is uneven right now, I'll fix that.

Before I go further with this... would you consider this change I just put up? 

https://github.com/rapidsai/shared-workflows/pull/177

Based on my experience here and in #166, I think it's worth taking a minute to re-sorting these matrices so they're easier to review.

Reply:
Alright @bdice whenever you have time, could you please take another look?

(and @msarahan @AyodeAwe @KyleFromNVIDIA if you'd like another chance to review now that changes have been made)

By my count, on the current state of this branch we have the following mix:

| workflow                            | python version | amd64 | arm64 | total |
|:-------------------------:|:--------------:|:--------:|:------:|:-----:|
| conda-python-build         | 3.9                    |  2           | 2         | 4       |
| conda-python-build         | 3.10                    |  2           | 2         | 4       |
| conda-python-build         | 3.11                    |  2           | 2         | 4       |
| conda-python-tests (pr) | 3.9                     | 1            | 0          | 1        | 
| conda-python-tests (pr) | 3.10                   | 1             | 1         | 2        |
| conda-python-tests (pr) | 3.11                   | 1             | 0         | 1         |
| conda-python-tests (nightly) | 3.9            | 2            | 1          | 3        | 
| conda-python-tests (nightly) | 3.10           | 2             | 1         | 3        |
| conda-python-tests (nightly) | 3.11            | 1             | 2        | 3        |
| wheel-build         | 3.9                    |  4           | 2         | 6       |
| wheel-build         | 3.10                    |  4           | 2         | 6       |
| wheel-build         | 3.11                    |  4           | 2         | 6       |
| wheel-test (pr) | 3.9                     | 1            | 0          | 1        | 
| wheel-test (pr) | 3.10                   | 0             | 1         | 1        |
| wheel-test (pr) | 3.11                   | 1             | 0         | 1         |
| wheel-test (nightly) | 3.9            | 2            | 1          | 3        | 
| wheel-test (nightly) | 3.10           | 2             | 2         | 4        |
| wheel-test (nightly) | 3.11            | 2             | 1        | 3        |

And the `conda-python-tests (pr)` and `wheel-test (pr)` both have the combination "amd64 + latest Python + latest CUDA".

I've still preserved the same absolute number of *test* jobs as `branch-24.04`.

I've also updated the description, with tests from these two PRs:

* https://github.com/rapidsai/rmm/pull/1469
* https://github.com/rapidsai/cudf/pull/15111